src/test/java/org.spongepowered.mod/test/integration/CreeperTest.java

```diff
@@ -0,0 +1,161 @@
+import org.spongepowered.api.entity.living.player.Player;
+import org.spongepowered.api.entity.living.player.gamemode.GameModes;
+import org.spongepowered.api.event.EventListener;
+import org.spongepowered.api.event.block.ChangeBlockEvent;
+import org.spongepowered.api.event.cause.EventContextKeys;
+import org.spongepowered.api.event.entity.MoveEntityEvent;
+import org.spongepowered.api.event.entity.SpawnEntityEvent;
+import org.spongepowered.api.event.entity.explosive.PrimeExplosiveEvent;
+import org.spongepowered.api.item.ItemTypes;
+import org.spongepowered.api.item.inventory.ItemStack;
+import org.spongepowered.api.item.inventory.entity.Hotbar;
+import org.spongepowered.api.item.inventory.query.QueryOperationTypes;
+import org.spongepowered.mctester.internal.BaseTest;
+import org.spongepowered.mctester.internal.event.StandaloneEventListener;
+import org.spongepowered.mctester.api.junit.MinecraftRunner;
+import org.spongepowered.mctester.junit.TestUtils;
+import org.spongepowered.mctester.api.WorldOptions;
+
+@RunWith(MinecraftRunner.class)
+@WorldOptions(deleteWorldOnSuccess = true)
+public class CreeperTest extends BaseTest {
+
+    public CreeperTest(TestUtils testUtils) {
+        super(testUtils);
+    }
+
+    @Test
+    public void explodeCreeper() throws Throwable {
+        this.testUtils.runOnMainThread(() -> {
+            Player player = this.testUtils.getThePlayer();
+            player.offer(Keys.GAME_MODE, GameModes.CREATIVE);
+
+            ((Hotbar) player.getInventory().query(QueryOperationTypes.INVENTORY_TYPE.of(Hotbar.class))).setSelectedSlotIndex(0);
+            player.setItemInHand(HandTypes.MAIN_HAND, ItemStack.builder().itemType(ItemTypes.SPAWN_EGG).quantity(1).add(Keys.SPAWNABLE_ENTITY_TYPE, EntityTypes.CREEPER).build());
+        });
+
+        testUtils.waitForInventoryPropagation();
+
+        // Look at the ground two blocks in the z direction
+        Vector3d targetPos = this.testUtils.runOnMainThread(() -> CreeperTest.this.testUtils.getThePlayer().getLocation().getPosition().add(0, -1, 2));
+        client.lookAt(targetPos);
+
+        final Creeper[] creeper = new Creeper[1];
+
+        this.testUtils.listen(new StandaloneEventListener<>(SpawnEntityEvent.class, (SpawnEntityEvent event) -> {
+            if (event.getEntities().stream().noneMatch(e -> e.getType().equals(EntityTypes.CREEPER))) {
+                return;
+            }
+
+            assertThat(event.getEntities(), hasSize(1));
+            creeper[0] = (Creeper) event.getEntities().get(0);
+
+            assertTrue("Cause doesn't contain player: " + event.getCause(), event.getCause().contains(CreeperTest.this.testUtils.getThePlayer()));
+            assertTrue("Cause doesn't contain correct item: " + event.getCause(),
+                    event.getCause().getContext().get(EventContextKeys.USED_ITEM).map(i -> i.getType().equals(ItemTypes.SPAWN_EGG)).orElse(false)
+            );
+
+            Sponge.getEventManager().unregisterListeners(this);
```

Comment:
Is the game instance recycled after each test? (not world instance) If yes you don't need to unregister the listener.

Reply:
No - one Minecraft client is used throughout all of the tests.=